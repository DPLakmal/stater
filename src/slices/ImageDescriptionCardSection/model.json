{
  "id": "image_description_card_section",
  "type": "SharedSlice",
  "name": "ImageDescriptionCardSection",
  "description": "ImageDescriptionCardSection",
  "variations": [
    {
      "id": "default",
      "name": "Default",
      "docURL": "...",
      "version": "initial",
      "description": "Default",
      "imageUrl": "",
      "primary": {
        "top_padding": {
          "type": "Select",
          "config": {
            "label": "Top Padding",
            "placeholder": "",
            "options": ["Full", "Half", "None"],
            "default_value": "Full"
          }
        },
        "bottom_padding": {
          "type": "Select",
          "config": {
            "label": "Bottom Padding",
            "placeholder": "",
            "options": ["Full", "Half", "None"],
            "default_value": "Full"
          }
        },
        "background_color": {
          "type": "Color",
          "config": {
            "label": "Background Color",
            "placeholder": ""
          }
        },
        "title": {
          "type": "StructuredText",
          "config": {
            "label": "Title",
            "placeholder": "",
            "allowTargetBlank": true,
            "multi": "heading1,heading2,heading3,heading4,heading5,heading6,strong,em"
          }
        },
        "description": {
          "type": "StructuredText",
          "config": {
            "label": "Description",
            "placeholder": "",
            "allowTargetBlank": true,
            "multi": "paragraph,heading6,strong,em,hyperlink"
          }
        },
        "cards": {
          "type": "Group",
          "config": {
            "label": "Cards",
            "repeat": true,
            "fields": {
              "right_aligned_image": {
                "type": "Boolean",
                "config": {
                  "label": "Right Aligned Image",
                  "placeholder_false": "false",
                  "placeholder_true": "true",
                  "default_value": false
                }
              },
              "image": {
                "type": "Image",
                "config": {
                  "label": "Image",
                  "constraint": {
                    "width": 542,
                    "height": 330
                  },
                  "thumbnails": []
                }
              },
              "icon": {
                "type": "Image",
                "config": {
                  "label": "Icon",
                  "constraint": {
                    "width": 56,
                    "height": 56
                  },
                  "thumbnails": []
                }
              },
              "title": {
                "type": "StructuredText",
                "config": {
                  "label": "Title",
                  "placeholder": "",
                  "allowTargetBlank": true,
                  "multi": "heading1,heading2,heading3,heading4,heading5,heading6,strong,em,hyperlink"
                }
              },
              "description": {
                "type": "StructuredText",
                "config": {
                  "label": "Description",
                  "placeholder": "",
                  "allowTargetBlank": true,
                  "multi": "paragraph,heading6,strong,em,hyperlink"
                }
              }
            }
          }
        }
      },
      "items": {}
    },
    {
      "id": "singleCard",
      "name": "Single Card",
      "docURL": "...",
      "version": "initial",
      "description": "Default",
      "imageUrl": "",
      "primary": {
        "top_padding": {
          "type": "Select",
          "config": {
            "label": "Top Padding",
            "placeholder": "",
            "options": ["Full", "Half", "None"],
            "default_value": "Full"
          }
        },
        "bottom_padding": {
          "type": "Select",
          "config": {
            "label": "Bottom Padding",
            "placeholder": "",
            "options": ["Full", "Half", "None"],
            "default_value": "Full"
          }
        },
        "background_color": {
          "type": "Color",
          "config": {
            "label": "Background Color",
            "placeholder": ""
          }
        },
        "title": {
          "type": "StructuredText",
          "config": {
            "label": "Title",
            "placeholder": "",
            "allowTargetBlank": true,
            "multi": "heading1,heading2,heading3,heading4,heading5,heading6,strong,em"
          }
        },
        "description": {
          "type": "StructuredText",
          "config": {
            "label": "Description",
            "placeholder": "",
            "allowTargetBlank": true,
            "multi": "paragraph,heading6,strong,em,hyperlink"
          }
        },
        "image": {
          "type": "Image",
          "config": {
            "label": "Image",
            "constraint": {
              "width": 414,
              "height": 392
            },
            "thumbnails": []
          }
        },
        "right_aligned_image": {
          "type": "Boolean",
          "config": {
            "label": "Right Aligned Image",
            "placeholder_false": "false",
            "placeholder_true": "true",
            "default_value": false
          }
        }
      },
      "items": {}
    }
  ]
}
