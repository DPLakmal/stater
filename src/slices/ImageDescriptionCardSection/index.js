import Main from "./variants/Main";
import SingleCard from "./variants/SingleCard";

/**
 * @typedef {import("@prismicio/client").Content.ImageDescriptionCardSectionSlice} ImageDescriptionCardSectionSlice
 * @typedef {import("@prismicio/react").SliceComponentProps<ImageDescriptionCardSectionSlice>} ImageDescriptionCardSectionProps
 * @param {ImageDescriptionCardSectionProps}
 */
const ImageDescriptionCardSection = ({ slice }) => {
  return (
    <section
      data-slice-type={slice.slice_type}
      data-slice-variation={slice.variation}
      style={{ backgroundColor: slice.primary.background_color }}
    >
      {slice.variation === "default" ? <Main slice={slice} /> : null}
      {slice.variation === "singleCard" ? <SingleCard slice={slice} /> : null}
    </section>
  );
};

export default ImageDescriptionCardSection;
