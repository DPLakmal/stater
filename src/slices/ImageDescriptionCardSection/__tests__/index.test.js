import "@testing-library/jest-dom";
import simplifyMockData from "@/utils/prismic/simplifyMockData";
import mocks from "../mocks.json";
import ImageDescriptionCardSection from "../index";
import { render } from "@testing-library/react";
import ImageDescriptionCardSection from "..";

describe("ImageDescriptionCard Section", () => {
  // top padding
  it("first variation with full top padding renders correctly", () => {
    let data = simplifyMockData(mocks[0]);
    data.primary.top_padding = "Full";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("first variation with half top padding renders correctly", () => {
    let data = simplifyMockData(mocks[0]);
    data.primary.top_padding = "Half";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("first variation without top padding renders correctly", () => {
    let data = simplifyMockData(mocks[0]);
    data.primary.top_padding = "None";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  // bottom padding
  it("first variation with full bottom padding renders correctly", () => {
    let data = simplifyMockData(mocks[0]);
    data.primary.bottom_padding = "Full";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("first variation with half bottom padding renders correctly", () => {
    let data = simplifyMockData(mocks[0]);
    data.primary.bottom_padding = "Half";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("first variation without bottom padding renders correctly", () => {
    let data = simplifyMockData(mocks[0]);
    data.primary.bottom_padding = "None";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });

  /* second variation */

  // top padding
  it("second variation with full top padding renders correctly", () => {
    let data = simplifyMockData(mocks[1]);
    data.primary.top_padding = "Full";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("second variation with half top padding renders correctly", () => {
    let data = simplifyMockData(mocks[1]);
    data.primary.top_padding = "Half";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("second variation without top padding renders correctly", () => {
    let data = simplifyMockData(mocks[1]);
    data.primary.top_padding = "None";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  // bottom padding
  it("second variation with full bottom padding renders correctly", () => {
    let data = simplifyMockData(mocks[1]);
    data.primary.bottom_padding = "Full";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("second variation with half bottom padding renders correctly", () => {
    let data = simplifyMockData(mocks[1]);
    data.primary.bottom_padding = "Half";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("second variation without bottom padding renders correctly", () => {
    let data = simplifyMockData(mocks[1]);
    data.primary.bottom_padding = "None";
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  //image align toggle
  it("second variation with left aligned image renders correctly", () => {
    const data = simplifyMockData(mocks[1]);
    data.primary.right_aligned_image = false;
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
  it("second variation with right aligned image renders correctly", () => {
    const data = simplifyMockData(mocks[1]);
    data.primary.right_aligned_image = true;
    const { container } = render(<ImageDescriptionCardSection slice={data} />);
    expect(container).toMatchSnapshot();
  });
});
