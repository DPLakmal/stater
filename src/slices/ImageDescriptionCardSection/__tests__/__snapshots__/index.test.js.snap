// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDescriptionCard Section first variation with full bottom padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="default"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[40px] md:pt-[80px] xl:pt-[120px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section first variation with full top padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="default"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[40px] md:pt-[80px] xl:pt-[120px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section first variation with half bottom padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="default"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pb-[20px] md:pb-[40px] xl:pb-[60px] pt-[40px] md:pt-[80px] xl:pt-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section first variation with half top padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="default"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[20px] md:pt-[40px] xl:pt-[60px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section first variation without bottom padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="default"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pb-0 pt-[40px] md:pt-[80px] xl:pt-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section first variation without top padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="default"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-0 pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
        <div
          class="mb-[30px] last:mb-0 lg:mb-0 last:lg:pb-0 lg:py-[40px] max-w-[580px] md:max-w-[680px] lg:max-w-none mx-auto lg:flex-row-reverse lg:flex lg:flex-nowrap lg:items-center lg:gap-[40px]"
        >
          <div
            class="mb-[14px]  lg:w-full"
          >
            <div
              class="flex items-center gap-[10px] mb-[8px] lg:mb-[20px] lg:gap-[14px]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation with full bottom padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[40px] md:pt-[80px] xl:pt-[120px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation with full top padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[40px] md:pt-[80px] xl:pt-[120px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation with half bottom padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pb-[20px] md:pb-[40px] xl:pb-[60px] pt-[40px] md:pt-[80px] xl:pt-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation with half top padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[20px] md:pt-[40px] xl:pt-[60px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation with left aligned image renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[40px] md:pt-[80px] xl:pt-[120px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation with right aligned image renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-[40px] md:pt-[80px] xl:pt-[120px] pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation without bottom padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pb-0 pt-[40px] md:pt-[80px] xl:pt-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`ImageDescriptionCard Section second variation without top padding renders correctly 1`] = `
<div>
  <section
    data-slice-variation="singleCard"
    style="background-color: rgb(244, 248, 255);"
  >
    <div
      class="px-[16px] md:px-[40px] xl:px-[80px] pt-0 pb-[40px] md:pb-[80px] xl:pb-[120px] "
    >
      <div
        class="mx-auto w-full max-w-[1280px] "
      >
        <div
          class="mx-auto max-w-[550px] md:max-w-[600px] lg:max-w-none lg:flex lg:flex-nowrap lg:items-center lg:gap-[38px] lg:justify-center lg:flex-row-reverse"
        >
          <div
            class="mb-[24px] w-full lg:max-w-[612px]"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
